### PR TITLE
Some more cleanup suggested by cargo clippy

### PR DESCRIPTION
**Pull request by crawford**
This is a follow-on to https://github.com/smoltcp-rs/smoltcp/pull/395. I still have some more changes locally, but I can follow up with another pull request once I get them cleaned up a bit.

### CODE REVIEW

File: src/wire/icmpv6.rs [hunk 1]
```diff
@@ -68,6 +69,7 @@ impl Message {
     ///
     /// [MLD]: https://tools.ietf.org/html/rfc3810
     pub fn is_mld(&self) -> bool {
+        #[allow(clippy::match_like_matches_macro)]
```
Comment (Dirbaio) [inline src/wire/icmpv6.rs]: Given that our MRSV is 1.36 and `matches!` is not supported there, it'd make more sense to allow it for the entire crate (`#![allow(clippy::match_like_matches_macro)]` in `lib.rs`)

File: src/wire/icmpv4.rs [hunk 2]
```diff
@@ -366,6 +366,7 @@ impl<T: AsRef<[u8]>> AsRef<[u8]> for Packet<T> {
 
 /// A high-level representation of an Internet Control Message Protocol version 4 packet header.
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[allow(clippy::manual_non_exhaustive)]
```
Comment (Dirbaio) [inline src/wire/icmpv4.rs]: I'd update these to use  `#[non_exhaustive]`, instead of adding an `#[allow]` for the old way:  #397

**Review by Dirbaio — CHANGES_REQUESTED** [review]

File: src/wire/icmpv4.rs (hunk 2 above)
Comment (crawford) [reply src/wire/icmpv4.rs]: That was introduced in Rust 1.40 and I didn't want to bump the minimum version.

Comment (Dirbaio) [reply src/wire/icmpv4.rs]: Oh okay, then I don't know what's up with #397 . Could you change it to a crate-wide allow then? Thank you!

**Comment by Dirbaio** [comment]
Very nice! Thanks!

**Pull request merged**